### DISCUSSION TITLE
How to make a frontend list view have filters like the backend list view

### DISCUSSION THREAD

**Discussion by SimonvanDoorne** [joomengine/Joomla-Component-Builder#633] [Q&A] [answered]
Hi there,

I'm making a list view on the frontend and i'm struggling to find an easy way to add filters to the page. Example, i have this listview in the backend:
![image](https://user-images.githubusercontent.com/54291562/103376311-9b146e00-4adc-11eb-8dfb-b4bfc31b9673.png)
On this page i added the filter 'Klasse' to the page using the new filter function in the admin view fields (super easy to use!), but now i want the same happening to the frontend view i made. Currently the page looks like this:
![image](https://user-images.githubusercontent.com/54291562/103376581-4d4c3580-4add-11eb-9b2c-5088de5c0051.png)
I want to add the same filters i have on the backend to the frontend, but i've not figured out a way to easily do this in JCB. 

I may be overlooking some simple solution, but i've looked in the view.html.php of the page and it doesn't look like there are any functions that would suggest that there is a pre built filter function JCB added. I have used Component-Creator before and in that system the filters you set are created for both front- and backend. Because of working with CC i know a bit on how to use the filtering system in Joomla, so i went looking in my project and i don't see a frontend XML file that is created when i'm compiling/installing my component.

I hope this makes clear what i'm trying to figure out, if not i'd be more than happy to supply more details!
Many thanks in advance!

**Comment by saman222** [comment]
Hi
You can watch this JCB series by mohamed abdelaziz, it was very helpful for me
https://www.youtube.com/user/MrMabdelaziz77

**Reply by TLWebdesign** [reply]
> Hi
> You can watch this JCB series by mohamed abdelaziz, it was very helpful for me
> https://www.youtube.com/user/MrMabdelaziz77

If you want a joomla only solution, this video is awesome!

**Comment by Llewellynvdm** [comment]
I have found the front to work better with libraries like [FooTable](https://fooplugins.github.io/FooTable/) which is able to load more than 10 000 items per page load with almost no delay. This library has its own filter system that is frankly way better than the Joomla back-end.

**Comment by Llewellynvdm** [comment, marked as answer]
But if that is not your cup of tea... the looking at the code use to make the back-end list view work should help you get it going. You will need to make sure that the correct JS code from Joomla, and the correct model, controller and view methods are lined-up as well. There has been request to make JCB move the list view to the front, like it does with the edit view.

Would you agree with that kind of feature request?

We are planning to add it as a pro member feature if we see enough interest.

**Reply by infotecnica** [reply]
i think it would be a great feature

**Reply by TLWebdesign** [reply]
+1 for automatic list views.

**Comment by MastersOfMedia** [comment]
That’s a good suggestion Llewellyn. Personally, I had not discovered FooTable yet. I have used Datatables (https://datatables.net/) and its even more powerful big brother Editor (https://editor.datatables.net/) extensively in the past.

Best regards,

Erik

**Reply by TLWebdesign** [reply]
I've actually been strugling with this and started on using footable but found it rather limiting by not using data attributes. Probably my knowledge was limiting it tho. Datatables works nice too but not for non table views. I ended up using isotope: https://isotope.metafizzy.co/filtering.html For working example see: https://connectmegroep.nl/ontdek-banen
